### DISCUSSION TITLE
default <Popover> tag should be fragment?

### DISCUSSION THREAD

**Discussion by ianstormtaylor** [tailwindlabs/headlessui#1121] [General]
Just an idea for an improvement. As far as I know a popover is most commonly a button (or some other click/hover target) and a panel:

```ts
<Popover>
  <Popover.Button />
  <Popover.Panel />
</Popover>
```

 But the current implementation has `<div>` as the default element for the root `<Popover>` component. I think the default should be `React.Fragment` instead so that no wrapper is rendered?

**Comment by Jbmanllr** [comment]
Hello, im sorry i wont answer your question but rather ask you a question: where did you see an actual popup component? I only find 4 components listed on the github project page...

**Comment by ianstormtaylor** [comment]
@Jbmanllr check out the `dev` branch, where some newer components are being worked on.

**Comment by simonswiss** [comment]
Hey @ianstormtaylor!

The main reason the `Popover` parent renders as a `div` by default is that it's quite common to add a `relative` class, so that the `Popover.Panel` can be positioned absolutely in relation to that.

Not always needed, but it's a fairly common approach, which is why we've opted to have a `div` as the default.

Since the `Popover` wrapper itself doesn't have any particular roles or attributes, you can indeed render it as a `Fragment` when you have no use for that wrapping element!

Going to convert this to a discussion 👍
